lib/sensu/transport/redis.rb

```diff
@@ -0,0 +1,307 @@
+require "em-redis-unified"
+
+require File.join(File.dirname(__FILE__), "base")
+
+module Sensu
+  module Transport
+    class Redis < Base
+
+      # The Redis keyspace to use for the transport.
+      REDIS_KEYSPACE = "transport"
+
+      def initialize
+        @options = {}
+        @connections = {}
+        super
+      end
+
+      # Redis transport connection setup. This method sets `@options`,
+      # creates a named Redis connection "redis", and starts a
+      # connection monitor.
+      #
+      # @param options [Hash, String]
+      def connect(options={})
+        @options = options || {}
+        redis_connection("redis")
+        monitor_connections
+      end
+
+      # Reconnect to the Redis transport. The Redis connections used
+      # by the transport have auto-reconnect disabled; if a single
+      # connection is unhealthy, all connections are closed, the
+      # transport is reset, and new connections are made. If the
+      # transport is not already reconnecting to Redis, the
+      # `@before_reconnect` transport callback is called.
+      #
+      # @param force [Boolean] the reconnect.
+      def reconnect(force=false)
+        @before_reconnect.call unless @reconnecting
+        unless @reconnecting && !force
+          @reconnecting = true
+          close
+          reset
+          connect
+        end
+      end
+
+      # Indicates if ALL Redis connections are connected.
+      #
+      # @return [TrueClass, FalseClass]
+      def connected?
+        !@connections.empty? && @connections.values.all? do |connection|
+          connection.connected?
+        end
+      end
+
+      # Close ALL Redis connections.
+      def close
+        @connections.each_value do |connection|
+          connection.close
+        end
+      end
+
+      # Publish a message to the Redis transport. The transport pipe
+      # type determines the method of sending messages to consumers
+      # using Redis, either using PubSub or a list. The appropriate
+      # publish method is call for the pipe type given. The Redis
+      # transport ignores publish options.
+      #
+      # @param type [Symbol] the transport pipe type, possible values
+      #   are: :direct and :fanout.
+      # @param pipe [String] the transport pipe name.
```

Comment:
That's fair. I does represent an IO target across implementations. Just had to think about it for a minute. :)